Dockerfile

```diff
@@ -34,7 +35,15 @@ RUN make deps
 
 # Build
 USER root
-RUN CGO_ENABLED=1 GOOS=linux GOARCH=amd64 make clean model-registry
+
+# NOTE: The two instructions below are effectively equivalent to 'make clean build'
+# DO NOT REMOVE THE 'build/prepare' TARGET!!!
+# It ensures consitent repeatable Dockerfile builds
+
+# prepare the build in a separate layer
+RUN make clean build/prepare
+# compile separately to optimize multi-platform builds
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} make build/compile
```

Comment:
I believe this `CGO_ENABLED=0` is functional making it _also_ work on Mac/Podman as discussed fwiw

Reply:
Yeah, in the future if we switch to some dependency that need CGO, there are options like installing cross compilation supporting c libs.